collections_hierarchy/mixins.py

```diff
@@ -1,53 +1,105 @@
 class ComparableMixin(object):
     def __eq__(self, other):
-        pass
+        if self.data == other.data:
+            return True
+        return False
+
     def __ne__(self, other):
         # Relies in __eq__
-        pass
+        return not self.__eq__(other)
+        
 
 
 class SequenceMixin(object):
+
+    def __init__(self, counter=0):
+        self.counter = counter
+
     def __iter__(self):
-        pass
+        self.counter = 0
+        return self
 
     def __next__(self):
         """This method will rely on the get_elements() method of the
         concrete class.
         """
+        try:
+            self.counter
+        except AttributeError:
+            self.counter = 0
         if not hasattr(self, 'get_elements'):
             raise ValueError("get_elements method not found")
-        # Keep writing your code here
-        raise NotImplementedError()
-
+        data = self.get_elements()
+        try:
+            if isinstance(data[0], tuple):
+                return_val = data[self.counter][0]
+            if not isinstance(data[0], tuple):
+                return_val = data[self.counter]
+            self.counter += 1
+        except IndexError:
+            raise StopIteration
+        return return_val
+    
```

Comment:
Why did you use all the try-except cases. You could do the same with if statements. Or not?